tests/tracking/test_tracking.py

```diff
@@ -40,6 +40,43 @@ def test_create_experiment():
         assert exp_id is not None
 
 
+def test_get_experiment_by_id():
+    with pytest.raises(TypeError):
+        tracking.get_experiment_by_id()
+
+    with pytest.raises(Exception):
+        tracking.get_experiment_by_id(-1)
+
+    with pytest.raises(Exception):
+        tracking.get_experiment_by_id(None)
+
+    with temp_directory() as tmp_dir, mock.patch("mlflow.tracking._get_store") as get_store_mock:
+        get_store_mock.return_value = FileStore(tmp_dir)
+        exp_id = tracking.create_experiment("Some random experiment name %d" % random.randint(1, 1e6))
+        assert exp_id is not None
+        experiment = tracking.get_experiment_by_id(exp_id)
+        assert experiment.experiment_id == exp_id
+
+
+def test_get_experiment_by_name():
```

Comment:
Nit: we can now use the `tracking_uri_mock` to set our tracking URI to a tempdir, e.g.:

```
from tests.projects.utils import tracking_uri_mock
def test_get_experiment_by_name(tracking_uri_mock):
   # Do stuff, the tracking URI will be mocked to point to a temporary directory
```

Reply:
I have looked up other examples using tracking_uri_mock. It was a bit unclear. If you don't mind I have used the same structure as the unit test for creating experiment.